[test_sub_port_l2_forwarding] [test issue] TypeError: a bytes-like object is required, not 'str'

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/sonic-net/SONiC/wiki#report-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**
test sub_port_interfaces/test_sub_port_l2_forwarding.py fail over python3 issue:
**TypeError: a bytes-like object is required, not 'str'**
<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**
run test_sub_port_l2_forwarding.py

**Describe the results you received:**
```
def test_sub_port_l2_forwarding(apply_config_on_the_dut, duthosts, rand_one_dut_hostname, test_sub_port,
                                    generate_eth_packets, testbed_params, ptfadapter):
        """Verify sub port doesn't have L2 forwarding capability."""
    
        @contextlib.contextmanager
        def check_no_cpu_packets(duthost, port, packet_fingerprint):
            start_pcap = "tcpdump -i %s -w %s" % (port, PACKET_SAVE_PATH)
            stop_pcap = "pkill -f '%s'" % start_pcap
            start_pcap = "nohup %s &" % start_pcap
    
            duthost.shell(start_pcap)
            try:
                yield
            finally:
                time.sleep(1.0)
                duthost.shell(stop_pcap, module_ignore_errors=True)
    
            with tempfile.NamedTemporaryFile() as tmp_pcap:
                duthost.fetch(src=PACKET_SAVE_PATH, dest=tmp_pcap.name, flat=True)
                received_packets = sniff(offline=tmp_pcap.name)
    
            logging.debug("Packets received from port %s:", port)
            for i, pkt in enumerate(received_packets):
                logging.debug("%d: %s" % (i, utilities.dump_scapy_packet_show_output(pkt)))
    
            packets_with_fingerprint = [_ for _ in received_packets if packet_fingerprint in str(_)]
            pytest_assert(len(packets_with_fingerprint) == 0, "Received packets with fingerprint %s" % packet_fingerprint)
    
        def verify_no_packet_received(ptfadapter, ports, packet_fingerprint):
            for port in ports:
                for packet, _ in ptfadapter.dataplane.packet_queues[(0, port)]:
                    if packet_fingerprint in packet:
                        logging.error("Received packet with fingerprint '%s' on port %s: %s\n", port, packet_fingerprint,
                                      packet)
                        pytest.fail("Received packet on port %s" % port)
    
        duthost = duthosts[rand_one_dut_hostname]
        packets = generate_eth_packets
        ptf_ports_to_check = list(set(_["neighbor_ptf_index"] for _ in list(testbed_params.values())))
        ptfadapter.dataplane.flush()
        for packet in packets:
            with check_no_cpu_packets(duthost, test_sub_port, PACKET_PAYLOAD_FINGERPRINT):
                testutils.send(ptfadapter, testbed_params[test_sub_port]["neighbor_ptf_index"], packet, count=PACKET_COUNT)
                time.sleep(TIME_WAIT_AFTER_SENDING_PACKET)
>               verify_no_packet_received(ptfadapter, ptf_ports_to_check, PACKET_PAYLOAD_FINGERPRINT)
```
```
def verify_no_packet_received(ptfadapter, ports, packet_fingerprint):
        for port in ports:
            for packet, _ in ptfadapter.dataplane.packet_queues[(0, port)]:
>               if packet_fingerprint in packet:
E               TypeError: a bytes-like object is required, not 'str'
```
**Describe the results you expected:**
test passes without test issues

**Additional information you deem important:**
<!--
software/ASIC/Hardware Flatform version and info
-->
**Output of `show version`:**

```
SONiC Software Version: SONiC.202211_RC8.3-696a6a43b_Internal
Distribution: Debian 11.6
Kernel: 5.10.0-18-2-amd64
Build commit: 696a6a43b
Build date: Mon Mar 27 10:50:51 UTC 2023
Built by: sw-r2d2-bot@r-build-sonic-ci03-244

Platform: x86_64-mlnx_msn3800-r0
HwSKU: Mellanox-SN3800-D112C8
ASIC: mellanox
ASIC Count: 1
Serial Number: MT2208X03840
Model Number: MSN3800-CS2FOS
Hardware Revision: A3
Uptime: 13:38:54 up 14 min,  2 users,  load average: 1.35, 1.48, 1.00
Date: Tue 28 Mar 2023 13:38:54

Docker images:
REPOSITORY                                         TAG                               IMAGE ID       SIZE
docker-orchagent                                   202211_RC8.3-696a6a43b_Internal   84055ebb4c77   539MB
docker-orchagent                                   latest                            84055ebb4c77   539MB
docker-fpm-frr                                     202211_RC8.3-696a6a43b_Internal   83d5f57abddb   550MB
docker-fpm-frr                                     latest                            83d5f57abddb   550MB
docker-teamd                                       202211_RC8.3-696a6a43b_Internal   e6b58409a5c6   520MB
docker-teamd                                       latest                            e6b58409a5c6   520MB
docker-macsec                                      latest                            d76f633dda63   522MB
docker-syncd-mlnx                                  202211_RC8.3-696a6a43b_Internal   2bd090589266   939MB
docker-syncd-mlnx                                  latest                            2bd090589266   939MB
docker-platform-monitor                            202211_RC8.3-696a6a43b_Internal   4fcfc27b9bbe   942MB
docker-platform-monitor                            latest                            4fcfc27b9bbe   942MB
docker-sonic-telemetry                             202211_RC8.3-696a6a43b_Internal   be9f9e750657   802MB
docker-sonic-telemetry                             latest                            be9f9e750657   802MB
docker-snmp                                        202211_RC8.3-696a6a43b_Internal   62432cf85f40   549MB
docker-snmp                                        latest                            62432cf85f40   549MB
docker-eventd                                      202211_RC8.3-696a6a43b_Internal   855b7c3d7218   503MB
docker-eventd                                      latest                            855b7c3d7218   503MB
docker-dhcp-relay                                  latest                            c90d1c5967eb   513MB
docker-lldp                                        202211_RC8.3-696a6a43b_Internal   e33be345d3f2   546MB
docker-lldp                                        latest                            e33be345d3f2   546MB
docker-mux                                         202211_RC8.3-696a6a43b_Internal   e9f6fa7ef277   552MB
docker-mux                                         latest                            e9f6fa7ef277   552MB
docker-database                                    202211_RC8.3-696a6a43b_Internal   9cae58961a25   503MB
docker-database                                    latest                            9cae58961a25   503MB
docker-sonic-p4rt                                  202211_RC8.3-696a6a43b_Internal   77b20e5b758d   586MB
docker-sonic-p4rt                                  latest                            77b20e5b758d   586MB
docker-router-advertiser                           202211_RC8.3-696a6a43b_Internal   f3a99688e9df   503MB
docker-router-advertiser                           latest                            f3a99688e9df   503MB
docker-sflow                                       202211_RC8.3-696a6a43b_Internal   780973343beb   479MB
docker-sflow                                       latest                            780973343beb   479MB
docker-nat                                         202211_RC8.3-696a6a43b_Internal   83454bb6925c   481MB
docker-nat                                         latest                            83454bb6925c   481MB
docker-sonic-mgmt-framework                        202211_RC8.3-696a6a43b_Internal   20cfc12ab3c5   612MB
docker-sonic-mgmt-framework                        latest                            20cfc12ab3c5   612MB
```

--- a/tests/sub_port_interfaces/test_sub_port_l2_forwarding.py
+++ b/tests/sub_port_interfaces/test_sub_port_l2_forwarding.py
@@ -110,9 +110,9 @@ def test_sub_port_l2_forwarding(apply_config_on_the_dut, duthosts, rand_one_dut_
     def verify_no_packet_received(ptfadapter, ports, packet_fingerprint):
         for port in ports:
             for packet, _ in ptfadapter.dataplane.packet_queues[(0, port)]:
-                if packet_fingerprint in packet:
+                if packet_fingerprint in str(packet):
                     logging.error("Received packet with fingerprint '%s' on port %s: %s\n", port, packet_fingerprint,
-                                  packet)
+                                  str(packet))
                     pytest.fail("Received packet on port %s" % port)
 
     duthost = duthosts[rand_one_dut_hostname]

